itl/command-plugins.conf

```diff
@@ -549,6 +549,175 @@ object CheckCommand "http" {
 	vars.http_verbose = false
 }
 
+object CheckCommand "curl" {
+	import "ipv4-or-ipv6"
+
+	command = [ PluginDir + "/check_curl" ]
+
+	arguments += {
+		"-H" = {
+			value = "$curl_vhost$"
+			description = "Host name argument for servers using host headers (virtual host). Append a port to include it in the header (eg: example.com:5000)"
+		}
+		"-I" = {
+			value = "$curl_ip$"
+			description = "IP address or name (use numeric address if possible to bypass DNS lookup)."
+		}
+		"-p" = {
+			value = "$curl_port$"
+			description = "Port number (default: 80)"
+		}
+		"-S" = {
+			set_if = "$curl_tls$"
+			value = "$curl_tls_version$"
+			description = "Connect via SSL. Port defaults to 443. VERSION is optional, and prevents auto-negotiation"
+		}
+		"--sni" = {
+			set_if = "$curl_sni$"
+			description = "Enable SSL/TLS hostname extension support (SNI). Default if TLS version > 1.0"
+		}
+		"-C" = {
+			value = "$curl_certificate_valid_days_min$"
+			description = "Minimum number of days a certificate has to be valid."
+		}
+		"-J" = {
+			value = "$curl_client-certificate-file$"
```

Comment:
A bit confusing is the mix of underscores and hyphens. Looking at other ITL definitions doesn't show those for other CheckCommands. Maybe just use: `curl_client_certificate_file`

This mix is also presence in other arguments.

Reply:
Actually a good catch. Had forgotten about that. The dashes might actually break things within `icinga2`s macro logic.